Develop

kimani-ndegwa opened the pull request.

mrkimani-ndegwa committed:
(Chore) Adds designs folder with UML designs

kimani-ndegwa committed:
(Chore) Adds draft.mdj File

kimani-ndegwa committed:
(Chore) Adds designs.mdj File
(Chore) Adds PNG, JPG & SVG file formats to class diagrams

kimani-ndegwa committed:
Merge branch 'designs' of github.com:andela-kndegwa/CP1 into designs

kimani-ndegwa committed:
Merge pull request #1 from andela-kndegwa/designs

Designs

kimani-ndegwa committed:
(Chore) Adds 3 tests for rooms allowing them to fail
(Chore) Organises file directory appropriately for Amity
(Chore)  Basic Test Functionality

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
(Chore) Tests Passing

kimani-ndegwa committed:
(Chore) Tests Passing.
(Chore) Adds Travics CI intergration.

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop
(Chore) Adds Travis Badge

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop
(Chore) Adds badge at beginning

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
(Chore) Adds Tests for amity class
(Chore) Adds Amity as the main class

kimani-ndegwa committed:
(Chore) Adds Tests for amity class
(Chore) Adds Amity as the main class with and refactorsT

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop
Resolves conflict
Conflicts:
	Amity/Rooms/room.py
(Chore) Adds more tests for Rooms and refactors code on the same

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop
Resolves conflict
Conflicts:
	Amity/Rooms/room.py

mrkimani-ndegwa committed:
Resolves conflict

mrkimani-ndegwa committed:
Resolves conflict
(Chore) Refactors the checkoccupants method in room.py

mrkimani-ndegwa committed:
(Chore): Edits comments

kimani-ndegwa committed:
Adds more definition to UML

kimani-ndegwa committed:
(Feature)  Adds Models and Database Functions
(Chore) Refactors code and aligns functionality
(Feature) Creates test_amity.db

kimani-ndegwa committed:
(Chore) Adds scripts Nosest

kimani-ndegwa committed:
(Chore) Adds scripts Nosetest
(Chore) Makes CP1 A package

kimani-ndegwa committed:
(Chore) Adds Nostests to .travis to solveconflicts

kimani-ndegwa committed:
Adds Rooms

kimani-ndegwa committed:
(Chore)Adds Rooms

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
(Chore): Refactor code to pass tests
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
(Chore) Removes Python 3.4 in Travsi config

kimani-ndegwa committed:
(Chore) Tests database works

kimani-ndegwa committed:
(Chore) Adds functinality to classes

kimani-ndegwa committed:
(Chore) Adds functinality to classes
(Chore) Adds 3.4 to travis

kimani-ndegwa committed:
(Chore)Fix travis
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

kimani-ndegwa committed:
Merge branch 'develop' of github.com:andela-kndegwa/CP1 into develop

mrkimani-ndegwa committed:
(Chore) Adds ft-ui branch
(Feature) Adds intuitive UI with ui.py file

mrkimani-ndegwa committed:
(Chore) Adds setup.py with entry point feature

kimani-ndegwa committed:
Merge pull request #2 from andela-kndegwa/ft-ui

(Chore) Adds ft-ui branch

kimani-ndegwa committed:
Merge pull request #2 from andela-kndegwa/ft-ui

(Chore) Adds ft-ui branch
(Chore) Adds setup.py with amity command working

mrkimani-ndegwa committed:
Merge branch 'ft-docopt-interactivity' of github.com:andela-kndegwa/CP1 into ft-docopt-interactivity

mrkimani-ndegwa committed:
(Chore) Adds interactivity to Docopt Inital cmd command
(Bug) CMD commands dont parse properly yet

kimani-ndegwa committed:
(Chore) Adds Path

kimani-ndegwa committed:
(Chore) Adds Path

mrkimani-ndegwa committed:
(Chore) Refactros tests for the purposes of working with nosetests
(Bug) Docopt front end not working yet

mrkimani-ndegwa committed:
(Chore) Adds allocator.py

kimani-ndegwa committed:
(Chore) Resloves conflict for tests in test_room
Merge branch 'tests-refactor' of github.com:andela-kndegwa/CP1 into ft-docopt-interactivity

Conflicts:
	Amity/Rooms/room.py
	Amity/tests/test_room.py
	amity.py

kimani-ndegwa committed:
(Chore) Fixes the db and tests

Merge branch 'ft-docopt-interactivity' of github.com:andela-kndegwa/CP1 into ft-docopt-interactivity

Conflicts:
	Amity/Rooms/room.py
	Amity/tests/test_room.py

kimani-ndegwa committed:
Merge branch 'ft-docopt-interactivity' of github.com:andela-kndegwa/CP1 into ft-docopt-interactivity

Conflicts:
	Amity/Rooms/room.py
	Amity/tests/test_room.py

kimani-ndegwa committed:
(Chore) Resloves conflict
(Chore) Adds Tests for create person and room
(Feature) Random allocation working properly
Merge branch 'ft-docopt-interactivity' of github.com:andela-kndegwa/CP1 into ft-docopt-interactivity

Conflicts:
	Amity/Rooms/room.py
	Amity/tests/test_room.py

mrkimani-ndegwa committed:
(Feature)Add Person works
(Feature)Allocate Room functionality working
(Feature)Reallocate room fnctionality working
(Bug) Allocate room returns error periodically

mrkimani-ndegwa committed:
(Feature)Add Person works
(Feature)Allocate Room functionality working
(Feature)Reallocate room fnctionality working
(Bug)Fixed Allocate room returns error periodically

kimani-ndegwa committed:
(Chore) Resolves conflict

Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

Conflicts:
	Amity/actions.py

kimani-ndegwa committed:
(Chore) Adds rooms and people folders

kimani-ndegwa committed:
(Chore) Adds rooms and people folders

kimani-ndegwa committed:
(Chore) Resolves conflict
Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

Conflicts:
	Amity/people/staff.py
	Amity/rooms/room.py

kimani-ndegwa committed:
(Chore) Resolves conflict
Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

Conflicts:
	Amity/people/staff.py
	Amity/rooms/room.py

mrkimani-ndegwa committed:
Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

mrkimani-ndegwa committed:
(Feature) Reallocate room parly functioning. Buggy
Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

mrkimani-ndegwa committed:
Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

kimani-ndegwa committed:
Refactor

kimani-ndegwa committed:
(Feature) Create room works on docopt
(Feature) Create room works with inheritance
(Feature) Add person functioality intergrated
(Feature) Print allocations also works

kimani-ndegwa committed:
(Feature) Create room works on docopt
(Feature) Create room works with inheritance

mrkimani-ndegwa committed:
(Feature) Print allocation working
(Feature) Add Person working

Merge branch 'actions' of github.com:andela-kndegwa/CP1 into actions

Conflicts:
	Amity/actions.py
	Amity/amity.py
	Amity/app.py
	Amity/people/person.py

kimani-ndegwa committed:
(Chore) Uses OOP Functionallyfor the create rooms funtionality

mrkimani-ndegwa committed:
(Chore) Reorganises the models into using classes and less lists and dictioaries
(Feature) Add room works great
(Feature) creat_room intergrated into docopt functionality

mrkimani-ndegwa committed:
(Feature) Add Person works
(Bug) Add person needs to return friendly error when no rooms exist

kimani-ndegwa committed:
(Feature) Reallocate person functioning
(BugFix) Allocation to rooms less erratic less erratic

kimani-ndegwa committed:
(Feature) Reallocate person functioning
(BugFix) Allocation to rooms less erratic less erratic
(Chore) Removes arbitrary code and adds doc strings to methods in the amity class

kimani-ndegwa committed:
(Chore) Resolves conflict with random arbitrary code.
Merge branch 'staging' of github.com:andela-kndegwa/CP1 into staging

Conflicts:
	Amity/amity.py

kimani-ndegwa committed:
(Chore) Updates file structure to reflect deleted files.

kimani-ndegwa committed:
(Feature) Adds progresbar to quit method
(Feature) Refactor loads_people

kimani-ndegwa committed:
(Chore)Edits README.md file

kimani-ndegwa committed:
(Chore)Edits README.md file

mrkimani-ndegwa committed:
(Chore)Changes Folder from Amity to amity

mrkimani-ndegwa committed:
(Chore)Removes unwanted files and folders from file system

mrkimani-ndegwa committed:
(Solves problem with identifiers)

mrkimani-ndegwa committed:
(Chore) Modifies amity with a number of passing tests
(Feature) Reallocation works better

mrkimani-ndegwa committed:
(Chore)Adds test coverage after coveralls configuration
(Feature) Intergrates with coveralls.io

mrkimani-ndegwa committed:
(Chore) Adds coverage badge

mrkimani-ndegwa committed:
(Chore) Adds coverage badge

mrkimani-ndegwa committed:
(Feature) Save State Functioning Properly
(Feature) Load State Functioning tentatively.
(Chore) Housekeeping on git files

kimani-ndegwa committed:
(Feature) Load State functionality implemented
(Chore) House keeping for the same
(Chore) Adds coverage to requirments.txt

kimani-ndegwa committed:
(Feature) Load State functionality implemented
(Chore) House keeping for the same

kimani-ndegwa committed:
(Chore) Resolves commit

kimani-ndegwa committed:
(Fix)Merge branch 'staging' of github.com:andela-kndegwa/CP1 into staging

Conflicts:
	amity/amity.py
	amity/default_amity_db.sqlite

kimani-ndegwa committed:
(Fix) Add coveralls to requirements.txt

kimani-ndegwa committed:
(Chore) Replaces Mock with actual tests

kimani-ndegwa committed:
(Chore) Adds asciinema screencast functionality
(Feature) Confirms functionality of the commands

mrkimani-ndegwa committed:
(Chore) Stash changes before push

mrkimani-ndegwa committed:
(Chore) Adds comprehensive tests that use the Mock feature
(Chore) Removes arbitrary test.py file
Merge branch 'staging' of github.com:andela-kndegwa/CP1 into staging

Conflicts:
	amity/amity.py
	amity/default_amity_db.sqlite

mrkimani-ndegwa committed:
(Chore) Housekeeping to keep the app.py file clean
(Chore) Removes comments in the amity.py file

mrkimani-ndegwa committed:
(Feature) Adds Print Out to file feature
(Chore) More clean up for cleaner code

mrkimani-ndegwa committed:
(Feature) Ability to specify storage database functioning
(Chore) Extra Clean Up

mrkimani-ndegwa committed:
(Chore)Edits README.md File.

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore)Modifies README.md file

mrkimani-ndegwa committed:
(Chore) Updates requirements.txt file

kimani-ndegwa committed:
Merge pull request #4 from andela-kndegwa/devtest

Devtest

njirap commented on amity/test_amity.py:
```diff
@@ -0,0 +1,275 @@
+    '''
+
+    def test_amity_class_initialises_with_nothing(self):
+        amity = Amity()
```
you can put this in `def setUp(self):`

kimani-ndegwa replied:
Done

njirap commented on amity/test_amity.py:
```diff
@@ -0,0 +1,275 @@
+                         'Error. No rooms within system.')
+    # add_person method testing begins here
+
+    def test_returns_error_if_no_rooms_within_system(self):
```
this should return the users as unallocated people

kimani-ndegwa replied:
This is a different validation. I don't want to accumulate unallocated people within my system but instead return an error that prompts them to add rooms first.

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+        if room_type.upper() == 'O':
+            room_type = 'Office'
+        if room_type.upper() == 'L':
+            room_type = 'Living Space'
```
`full_room_names = {'O':'Office', 'L':'Living Space'}

full_room_names[room_type]`

kimani-ndegwa replied:
Done

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+                click.secho('%s --> %s ALREADY EXISTS.Please pick another name.'
+                            % (room_type, room_name.title()),
+                            fg='red', bold=True)
+                return 'Room already exists.'
```
`if [room for self.rooms if room.room_type == room_type.title() and room.room_name == room_name.title()]`

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+                            fg='red', bold=True)
+                return 'Room already exists.'
+        room_type = room_type.strip().title()
+        room_name = room_name.strip().title()
```
you can move these 2 lines to before you used in L62 and L63 so you don't have to repeat the same thing in the for loop..

kimani-ndegwa replied:
Done

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+        elif room_type == 'Living Space':
+            room = LivingSpace(room_name)
+            self.living_spaces['available'].append(room.room_name)
+        self.rooms.append(room)
```
this is not a necessary variable to have. It can be:
`[self.offices['available'] + [self.offices['unavailable'] + [self.livingspaces['available'] + [self.livingspacess['available']]`

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+                        file.write(occupant + ' ')
+                else:
+                    file.write('%s is empty' % room.room_name)
+            click.secho('Print out made to %s.txt' % filename, fg='green')
```
this logic can be improved by first building up the text then if there is a filename then print to file, otherwise colour it and print to terminal

kimani-ndegwa replied:
Done

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+        if type(first_name) != str or type(other_name) != str:
+            click.secho('Incorrect name type format.', fg='red')
+            return 'Wrong type for name.'
+        if first_name.isalpha() is False or other_name.isalpha() is False:
```
`if !first_name.isalpha() or !other_name.isalpha()`

kimani-ndegwa replied:
Done

njirap commented on amity/amity.py:
```diff
@@ -0,0 +1,599 @@
+            click.secho(
+                'A Staff member cannot be allocated accomodation.',
+                fg='red', bold=True)
+            return 'Staff cannot have wants accomodation of Y.'
```
Just assign the Staff to an office but tell them that they cannot have accommodation

njirap replied:
abstract allocate_people logic so that it is DRY and not repeated in the load_txt logic

kimani-ndegwa replied:
Done

njirap commented:
maybe also get the CI to work..?

kimani-ndegwa committed:
(Chore) Refactors code to allow staff to be allocated if accomodate

kimani-ndegwa committed:
(Chore) Stashes changes before squash

kimani-ndegwa committed:
(Chore) Breaks code for the add person functionality
(Feature) Allows Staff to be allocated an office regardless of whether they want accomodation

kimani-ndegwa committed:
refactor changes

kimani-ndegwa committed:
(Chore) Refactoring code file

kimani-ndegwa committed:
(Chore) Refactors add_person function into 3 separate functions and passes tests

kimani-ndegwa committed:
(Chore) Tweaks color for the purpose of displaying people names

mrkimani-ndegwa committed:
(BugFix) Fixes bug with the add person functionality.

mrkimani-ndegwa committed:
(Chore) Adds codacy and coverage badge also from codacy
(Chore) Removes ar.py file

mrkimani-ndegwa committed:
(Chore) Confirms delete of ar.py file

kimani-ndegwa committed:
Merge pull request #6 from andela-kndegwa/dev-refactor

Dev refactor

mrkimani-ndegwa committed:
(Chore) Adds testing information to README.md

mrkimani-ndegwa committed:
Merge branch 'dev-refactor' into develop

mrkimani-ndegwa committed:
(Chore) Corrects numbering on README.md
